Flaky Deconz test

This test failed on #32039 which had a change unrelated to Deconz.

https://dev.azure.com/home-assistant/Core/_build/results?buildId=28255&view=logs&j=67510a0b-4400-5e74-cb95-9572caf5ca54&t=f8fe93fb-5ebb-59cc-455e-c0f2d9ad3a42&l=238

```
    async def test_climate_devices(hass):
        """Test successful creation of sensor entities."""
        data = deepcopy(DECONZ_WEB_REQUEST)
        data["sensors"] = deepcopy(SENSORS)
        gateway = await setup_deconz_integration(hass, get_state_response=data)
>       assert "climate.thermostat" in gateway.deconz_ids
E       AssertionError: assert 'climate.thermostat' in {'binary_sensor.presence_sensor': '/sensors/2', 'sensor.thermostat_battery_level': '/sensors/1'}
E        +  where {'binary_sensor.presence_sensor': '/sensors/2', 'sensor.thermostat_battery_level': '/sensors/1'} = <homeassistant.components.deconz.gateway.DeconzGateway object at 0x7fe19228b7b8>.deconz_ids

tests/components/deconz/test_climate.py:69: AssertionError
```

--- a/homeassistant/components/deconz/binary_sensor.py
+++ b/homeassistant/components/deconz/binary_sensor.py
@@ -37,7 +37,6 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                     gateway.option_allow_clip_sensor
                     or not sensor.type.startswith("CLIP")
                 )
-                and sensor.deconz_id not in gateway.deconz_ids.values()
             ):
                 entities.append(DeconzBinarySensor(sensor, gateway))
 

--- a/homeassistant/components/deconz/climate.py
+++ b/homeassistant/components/deconz/climate.py
@@ -44,7 +44,6 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                     gateway.option_allow_clip_sensor
                     or not sensor.type.startswith("CLIP")
                 )
-                and sensor.deconz_id not in gateway.deconz_ids.values()
             ):
                 entities.append(DeconzThermostat(sensor, gateway))
 

--- a/homeassistant/components/deconz/light.py
+++ b/homeassistant/components/deconz/light.py
@@ -67,7 +67,7 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         entities = []
 
         for group in groups:
-            if group.lights and group.deconz_id not in gateway.deconz_ids.values():
+            if group.lights:
                 entities.append(DeconzGroup(group, gateway))
 
         async_add_entities(entities, True)

--- a/homeassistant/components/deconz/sensor.py
+++ b/homeassistant/components/deconz/sensor.py
@@ -68,7 +68,6 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
                     gateway.option_allow_clip_sensor
                     or not sensor.type.startswith("CLIP")
                 )
-                and sensor.deconz_id not in gateway.deconz_ids.values()
             ):
                 entities.append(DeconzSensor(sensor, gateway))
 

